client/dashboard/app/queries/site.ts

```diff
@@ -26,8 +26,25 @@ export const siteBySlugQuery = ( siteSlug: string ) => ( {
 } );
 
 export const siteByIdQuery = ( siteId: number ) => ( {
+	queryKey: [ 'site-by-id', siteId, SITE_FIELDS, SITE_OPTIONS ],
+	queryFn: async () => {
+		try {
+			return await fetchSite( siteId );
+		} catch ( e: any ) /* eslint-disable-line @typescript-eslint/no-explicit-any */ {
+			if ( e.error === 'unknown_blog' ) {
+				throw notFound();
+			}
+			throw e;
+		}
+	},
+} );
+
+export const siteQueryFilter = ( siteId: number ) => ( {
 	predicate: ( { queryKey, state }: Query ) => {
-		return queryKey[ 0 ] === 'site-by-slug' && ( state.data as Site )?.ID === siteId;
+		return (
+			( queryKey[ 0 ] === 'site-by-slug' || queryKey[ 0 ] === 'site-by-id' ) &&
+			( state.data as Site )?.ID === siteId
```

Comment:
Question: I'm a bit confused about this. If the query key is `site-by-slug` but it matches siteId, does it mean that there's a bug or the cache is being misused?

Reply:
It's because there are two queries:
-  `siteBySlugQuery()` stores the data at [ `site-by-slug`, site.slug, ... ].
    - this is used for fetching a site if we only have its slug. used for loading the current site in the hosting dashboard.
- `siteByIdQuery()` stores the data at [ `site-by-id`, site.ID, ... ].
    - this is used for fetching a site if we have its ID. used for loading staging site given its ID.

This filter will match both query keys given a site ID.